posthog/hogql_queries/web_analytics/pre_aggregated/query_builder.py

```diff
@@ -86,7 +86,7 @@ def _get_filters(self, table_name: str):
 
         return None
 
-    def get_date_ranges(self) -> tuple[str, str]:
+    def get_date_ranges(self, table_name: Optional[str] = None) -> tuple[str, str]:
```

Comment:
This is getting out of hand and could really use the ast and some helpers like the original `stats_table`. I think this was the most complex one as it required the bounces inner query with a separate group by so that is something but added to the refactor notes.